File: v20_flyweight/java/DrawingApp1.java
package v20_flyweight.java;

class Circle {
    private String color;
    private int x;
    private int y;
    private int radius;

    public Circle(String color) {
        this.color = color;
    }

    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    public void setRadius(int radius) {
        this.radius = radius;
    }

    public void draw() {
        System.out.printf("Drawing circle of color %s at position (%d, %d) with radius %d\n", color, x, y, radius);
    }
}

public class DrawingApp1 {
    public static void main(String[] args) {
        Circle redCircle = new Circle("red");
        redCircle.setX(10);
        redCircle.setY(10);
        redCircle.setRadius(50);
        redCircle.draw();

        Circle blueCircle = new Circle("blue");
        blueCircle.setX(20);
        blueCircle.setY(20);
        blueCircle.setRadius(60);
        blueCircle.draw();
    }
}

File: v20_flyweight/python/drawing_app1.py
class Circle:
    def __init__(self, color):
        self.color = color
        self.x = None
        self.y = None
        self.radius = None

    def set_x(self, x):
        self.x = x

    def set_y(self, y):
        self.y = y

    def set_radius(self, radius):
        self.radius = radius

    def draw(self):
        print(f'Drawing circle of color {self.color} at position ({self.x}, {self.y}) with radius {self.radius}')

class DrawingApp1:
    @staticmethod
    def main():
        red_circle = Circle("red")
        red_circle.set_x(10)
        red_circle.set_y(10)
        red_circle.set_radius(50)
        red_circle.draw()

        blue_circle = Circle("blue")
        blue_circle.set_x(20)
        blue_circle.set_y(20)
        blue_circle.set_radius(60)
        blue_circle.draw()

if __name__ == "__main__":
    DrawingApp1.main()


